vertx-web/src/main/java/io/vertx/ext/web/impl/HandlersCollection.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2023 Red Hat, Inc.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Apache License v2.0 which accompanies this distribution.
+ *
+ * The Eclipse Public License is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * The Apache License v2.0 is available at
+ * http://www.opensource.org/licenses/apache2.0.php
+ *
+ * You may elect to redistribute this code under either of these licenses.
+ */
+
+package io.vertx.ext.web.impl;
+
+import io.vertx.core.Handler;
+
+import java.util.ArrayList;
+import java.util.List;
+
+class HandlersCollection<E> {
+
+  private final List<Handler<E>> list;
+
+  HandlersCollection() {
+    this.list = new ArrayList<>();
+  }
+
+  int put(Handler<E> handler) {
+    list.add(handler);
+    return list.size() - 1;
+  }
+
+  boolean remove(int handlerID) {
+    if (handlerID < 0 || handlerID >= list.size()) {
+      return false;
+    }
+    return list.set(handlerID, null) != null;
+  }
+
+  void clear() {
+    list.clear();
+  }
+
+  void invokeInReverseOrder(E event) {
+    for (int i = list.size() - 1; i >= 0; i--) {
```

Comment:
In theory an handler could remove/add elements with virtually no harm given that we snapshot the expected size-limit of it just once in the for loop, but is it something we need to be protected against? if yes, we can use the modCount trick and throw ConcurrentModificationException, similarly to ArrayList, if not, I prefer it as it is.

Reply:
I don't think it is necessary because:

- if a handler is removed, it won't be invoked (null-check below)
- if a handler is added, it won't be invoked (id will be bigger than the size known when the method is invoked)
- it makes little sense to add/remove event handlers when the event occurs because headers sent, body sent and response sent events can, by definition, only happen once.